pilot/pkg/model/push_context.go

```diff
@@ -894,6 +895,16 @@ func (ps *PushContext) InitContext(env *Environment, oldPushContext *PushContext
 	ps.ServiceDiscovery = env.ServiceDiscovery
 	ps.IstioConfigStore = env.IstioConfigStore
 	ps.LedgerVersion = env.Version()
+	ps.Cleanup = func() {
+		l := env.GetLedger()
```

Comment:
nit: might be cleaner to just store the ledger in `PushContext` struct and make Cleanup a normal method?

Reply:
bootrstrap/configcontroller.go:264 needs access to it at the server level, which is why it is part of environment.  I could store it both places, I suppose.  Would that be better?

Reply:
I mean replace this with
```
ps.ledger := env.GetLedger()
....

func (ps *PushContext) Cleanup() {
...
}
```
```

Reply:
The cleanup function needs access to the ledger, which is not stored by the pushcontext outside of the InitContext function.

Reply:
I am suggesting you store it in the push context. Its not very standard to store a closure, rather than just having a standard close function